napari/_qt/_qapp_model/qactions/_file.py

```diff
@@ -0,0 +1,126 @@
+from typing import List
+
+from app_model.types import Action, KeyCode, KeyMod, StandardKeyBinding
+
+from ...._app_model.constants import CommandId, MenuGroup, MenuId
+from ....utils.translations import trans
+from ...qt_main_window import Window
+from ...qt_viewer import QtViewer
+
+
+def _restart(window: Window):
+    window._qt_window.restart()
+
+
+Q_FILE_ACTIONS: List[Action] = [
+    Action(
+        id=CommandId.DLG_OPEN_FILES,
+        title=CommandId.DLG_OPEN_FILES.title,
+        callback=QtViewer._open_files_dialog,
+        menus=[{'id': MenuId.MENUBAR_FILE, 'group': MenuGroup.NAVIGATION}],
+        keybindings=[StandardKeyBinding.Open],
+    ),
+    Action(
+        id=CommandId.DLG_OPEN_FILES_AS_STACK,
+        title=CommandId.DLG_OPEN_FILES_AS_STACK.title,
+        callback=QtViewer._open_files_dialog_as_stack_dialog,
+        menus=[{'id': MenuId.MENUBAR_FILE, 'group': MenuGroup.NAVIGATION}],
+        keybindings=[{'primary': KeyMod.CtrlCmd | KeyMod.Alt | KeyCode.KeyO}],
+    ),
+    Action(
+        id=CommandId.DLG_OPEN_FOLDER,
+        title=CommandId.DLG_OPEN_FOLDER.title,
+        callback=QtViewer._open_files_dialog,
+        menus=[{'id': MenuId.MENUBAR_FILE, 'group': MenuGroup.NAVIGATION}],
+        keybindings=[
+            {'primary': KeyMod.CtrlCmd | KeyMod.Shift | KeyCode.KeyO}
+        ],
+    ),
+    Action(
+        id=CommandId.DLG_SHOW_PREFERENCES,
+        title=CommandId.DLG_SHOW_PREFERENCES.title,
+        callback=Window._open_preferences_dialog,
+        menus=[{'id': MenuId.MENUBAR_FILE, 'group': '2_'}],
+        keybindings=[StandardKeyBinding.Preferences],
+    ),
+    # TODO!
+    # this action would conflict because having the same id as below... but
+    # we could check whether the args are the same and if not, then allow registration
+    # Action(
+    #     id=CommandId.DLG_SAVE_LAYERS,
+    #     title=CommandId.DLG_SAVE_LAYERS.title,
+    #     callback=QtViewer._save_layers_dialog,
+    #     kwargs={'selected': True},  # <<<<< TODO!
+    #     menus=[{'id': MenuId.MENUBAR_FILE, 'group': '3_'}],
+    #     keybindings=[StandardKeyBinding.Save],
+    #     enablement='num_selected_layers > 0'
+    # ),
```

Comment:
I am curious about the use of id here: 
It seems easier to define the two save layers as a separate command with their own ID, I don't see the benefit of forcing them to share the same ID immediately.

If the rationale is for the name convention between id and callback, this convention is broken in a few other places like TOGGLE_MENUBAR <-> Window._toggle_menubar_visible.

If the rationale is to reduce and compact similar callbacks, and all functions with the same callback should have the same ID, I would argue we can just drop ID and auto-generate them from the callback name and code hierarchy.